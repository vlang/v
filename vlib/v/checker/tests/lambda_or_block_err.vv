fn foo(callback fn (int) int) {
	dump([1, 2, 3].map(callback))
}

fn bar(a int) !int {
	return a
}

fn main() {
	foo(|i| bar(i))
	foo(|i| bar(i) or {})
	foo(|i| bar(i) or { 0 })
}
