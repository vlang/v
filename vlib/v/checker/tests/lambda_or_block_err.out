vlib/v/checker/tests/lambda_or_block_err.vv:10:10: error: cannot use `!int` as type `int` in return argument
    8 | 
    9 | fn main() {
   10 |     foo(|i| bar(i))
      |             ~~~~~~
   11 |     foo(|i| bar(i) or {})
   12 |     foo(|i| bar(i) or { 0 })
vlib/v/checker/tests/lambda_or_block_err.vv:11:17: error: expression requires a non empty `or {}` block
    9 | fn main() {
   10 |     foo(|i| bar(i))
   11 |     foo(|i| bar(i) or {})
      |                    ~~~~~
   12 |     foo(|i| bar(i) or { 0 })
   13 | }
